zebra-network/src/peer/handshake.rs

```diff
@@ -401,9 +401,24 @@ where
                             // query rather than a newly gossiped block.
                             //
                             // https://zebra.zfnd.org/dev/rfcs/0003-inventory-tracking.html#inventory-monitoring
-                            if hashes.len() == 1 {
-                                let hash = hashes[0];
-                                let _ = inv_collector.send((hash, addr));
+                            match hashes.as_slice() {
+                                [hash @ InventoryHash::Block(_)] => {
+                                    let _ = inv_collector.send((*hash, addr));
+                                }
+                                [hashes @ ..]
+                                    if hashes
+                                        .iter()
+                                        .any(|&hash| matches!(hash, InventoryHash::Tx(_))) =>
+                                {
+                                    for hash in hashes {
+                                        if matches!(hash, InventoryHash::Tx(_)) {
+                                            let _ = inv_collector.send((*hash, addr));
+                                        } else {
+                                            debug!(?hash, "ignored non Tx inventory hash")
+                                        }
+                                    }
+                                }
+                                ignored => debug!(?ignored, "ignored inventory advert"),
```

Comment:
Rather than logging a tracing event on ignored hashes, I think it'd be more useful to log events on the hashes we do register, either here or in the inventory registry (but probably not both?).  We could demote ignored inventory advertisements to trace level, I think.

Reply:
pushed in 2f48227

tested with

```
zebra/zebrad on  inv-adverts-fix [$] is 📦 v3.0.0-alpha.0 via 🦀 v1.47.0-nightly took 4s 
❯ cat ~/.config/zebrad.toml 
[tracing]
endpoint_addr = "127.0.0.1:3000"
filter = "info,zebra_network[inventory_filter]=debug"
```

```
Sep 08 11:04:35.454 DEBUG peer{addr=209.250.241.75:8233}:inventory_filter: zebra_network::peer::handshake: registering Tx inventory hash hash=Tx(TransactionHash("ad74506eecad96c3e1dc92fbd867f7c7223c13c3139871ebfad1bed2ab4fa6b8"))
Sep 08 11:04:36.042 DEBUG peer{addr=47.93.17.158:8233}:inventory_filter: zebra_network::peer::handshake: registering Tx inventory hash hash=Tx(TransactionHash("ad74506eecad96c3e1dc92fbd867f7c7223c13c3139871ebfad1bed2ab4fa6b8"))
Sep 08 11:04:36.372 DEBUG peer{addr=3.112.252.179:8233}:inventory_filter: zebra_network::peer::handshake: registering Tx inventory hash hash=Tx(TransactionHash("ad74506eecad96c3e1dc92fbd867f7c7223c13c3139871ebfad1bed2ab4fa6b8"))
Sep 08 11:04:36.888 DEBUG peer{addr=34.87.37.112:8233}:inventory_filter: zebra_network::peer::handshake: registering Tx inventory hash hash=Tx(TransactionHash("ad74506eecad96c3e1dc92fbd867f7c7223c13c3139871ebfad1bed2ab4fa6b8"))
```